commands/utils/quick-pick-azure.ts

```diff
@@ -0,0 +1,168 @@
+    return desiredSku.data;
+}
+
+export async function quickPickSubscription(): Promise<Subscription> {
+    const subscriptions = AzureUtilityManager.getInstance().getFilteredSubscriptionList();
+    if (subscriptions.length === 0) {
+        vscode.window.showErrorMessage("You do not have any subscriptions. You can create one in your Azure portal", "Open Portal").then(val => {
+            if (val === "Open Portal") {
+                opn('https://portal.azure.com/');
+            }
+        });
+        throw new Error('User has no azure subscriptions');
+    }
+    let quickPickSubList = subscriptions.map(sub => <IAzureQuickPickItem<Subscription>>{ label: sub.displayName, data: sub });
+    let desiredSub = await ext.ui.showQuickPick(quickPickSubList, {
+        'canPickMany': false,
+        'placeHolder': 'Choose a subscription to use'
+    });
+    return desiredSub.data;
+}
+
+export async function quickPickLocation(subscription: Subscription): Promise<string> {
+    let locations: Location[] = await AzureUtilityManager.getInstance().getLocationsBySubscription(subscription);
+    let quickPickLocList = locations.map(loc => <IAzureQuickPickItem<Location>>{ label: loc.displayName, data: loc });
+
+    quickPickLocList.sort((loc1, loc2): number => {
+        return loc1.data.displayName.localeCompare(loc2.data.displayName);
+    });
+
+    let desiredLocation: IAzureQuickPickItem<Location> = await ext.ui.showQuickPick(quickPickLocList, {
+        'canPickMany': false,
+        'placeHolder': 'Choose a location to use'
+    });
+    return desiredLocation.label;
+}
+
+export async function quickPickResourceGroup(canCreateNew?: boolean, subscription?: Subscription): Promise<ResourceGroup> {
+    let resourceGroups = await AzureUtilityManager.getInstance().getResourceGroups(subscription);
+    let resourceGroupNames: string[] = [];
+
+    if (canCreateNew) { resourceGroupNames.push('+ Create new resource group'); }
+    for (let resGroupName of resourceGroups) {
+        resourceGroupNames.push(resGroupName.name);
+    }
+
+    let resourceGroupName = await vscode.window.showQuickPick(resourceGroupNames, {
+        'canPickMany': false,
+        'placeHolder': 'Choose a resource group to use'
+    });
+
+    let resourceGroup: ResourceGroup;
+    if (canCreateNew && resourceGroupName === resourceGroupNames[0]) {
+        if (!subscription) {
+            subscription = await quickPickSubscription();
+        }
+        const loc = await quickPickLocation(subscription);
+        resourceGroup = await createNewResourceGroup(loc, subscription);
+    } else {
+        resourceGroup = resourceGroups.find(resGroup => { return resGroup.name === resourceGroupName; });
+    }
+    return resourceGroup;
+}
+
+/** Requests confirmation for an action and returns true only in the case that the user types in yes
+ * @param yesOrNoPrompt Should be a yes or no question
+ */
+export async function confirmUserIntent(yesOrNoPrompt: string): Promise<boolean> {
```

Comment:
Rather than returning a Boolean, throw a UserCancelledError error (from vscode-azureextensionui) if they don't confirm. The reason for this is that our telemetry code will think the command excecuted and succeeded if you simply return when they say no.  Throwing user canceled will let us know instead that the user, well, canceled.

Reply:
I was wondering about that, perfect